index.d.ts

```diff
@@ -0,0 +1,170 @@
+declare module 'fluture' {
+  interface ResolveFunction<R> {
+    (value: R): void
+  }
+  interface RejectFunction<L> {
+    (error: L): void
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  export abstract class IFuture<L, R> {
+    fork(reject: RejectFunction<L>, resolve: ResolveFunction<R>): Cancel
+  }
+
+  export function after<L, R>(duration: number, value: R): IFuture<L, R>
+
+  export function and<L, R>(
+    left: IFuture<L, any>,
+    right: IFuture<L, R>
+  ): IFuture<L, R>
+
+  export function ap<L, R, VR>(
+    apply: IFuture<L, (value: VR) => R>,
+    value: IFuture<L, VR>
+  ): IFuture<L, R>
+
+  export function attempt<L, R>(f: () => L): IFuture<L, R>
+
+  export function bimap<A, B, AA, BB>(
+    left: (a: A) => AA,
+    right: (a: B) => BB
+  ): IFuture<AA, any>
+
+  export function both<L, A, B>(
+    left: IFuture<L, A>,
+    right: IFuture<L, B>
+  ): IFuture<L, [A, B]>
+
+  export function cache<L, R>(source: IFuture<L, R>): IFuture<L, R>
+
+  export function chain<L, R, RR>(
+    mapper: (value: R) => IFuture<L, RR>,
+    source: IFuture<L, R>
+  ): IFuture<L, RR>
+
+  export function done<L, R>(
+    fn: (cb: (err: L | null, value: R) => void) => void
+  ): IFuture<L, R>
```

Comment:
It seems you confused this function for `Future.node`. Instead it be somewhat like this:

```ts
done<L, R>(
  callback: (err: L | null, value: R?) => void,
  source: IFuture<L, R>
): Cancel
```